src/project.c

```diff
@@ -562,24 +562,40 @@ int  addlinkvertex(Slink *link, double x, double y)
     static int CHUNKSIZE = 5;
     int n;
     Pvertices vertices;
+    double* newPtr;
+
     if (link->Vertices == NULL)
     {
         vertices = (struct Svertices *) malloc(sizeof(struct Svertices));
         if (vertices == NULL) return 101;
         vertices->Npts = 0;
         vertices->Capacity = CHUNKSIZE;
         vertices->X = (double *) calloc(vertices->Capacity, sizeof(double));
+        if (vertices->X == NULL)
+        {
+            free(vertices);
+            return 101;
+        }
         vertices->Y = (double *) calloc(vertices->Capacity, sizeof(double));
+        if (vertices->Y == NULL)
+        {
+            free(vertices->X);
+            free(vertices);
+            return 101;
+        }
         link->Vertices = vertices;
     }
     vertices = link->Vertices;
     if (vertices->Npts >= vertices->Capacity)
     {
         vertices->Capacity += CHUNKSIZE;
-        vertices->X = realloc(vertices->X, vertices->Capacity * sizeof(double));
-        vertices->Y = realloc(vertices->Y, vertices->Capacity * sizeof(double));
+        newPtr = realloc(vertices->X, vertices->Capacity * sizeof(double));
```

Comment:
Use of `newPtr `is not needed since original line 582 tests if either `vertices->X` or` vertices->Y` is `NULL`.